src/pages/Login/components/User.js

```diff
@@ -0,0 +1,205 @@
+    /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i
+  );
+  const passwordRegex = new RegExp(
+    /^(?=.*[A-Za-z])(?=.*\d)(?=.*[~!@#$%^&*()+|=])[A-Za-z\d~!@#$%^&*()+|=]{8,16}$/
+  );
+
+  // 로그인 회원가입 폼 구분
+  const [isLogin, setIsLogin] = useState(title === '로그인');
+
+  // 입력된 데이터 유효성 검사
+  const isValid = isLogin
+    ? emailRegex.test(formData.id) && formData.password.length >= 5
+    : formData.name.length > 0 &&
+      emailRegex.test(formData.id) &&
+      formData.number.length > 0 &&
+      passwordRegex.test(formData.password);
+
+  // 입력되는 데이터 변경시
+  const onChangeFormData = e => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  //로그인 회원가입하기로 이동하기 클릭시
+  const handleUrlClick = () => {
+    setFormData({
+      id: '',
+      password: '',
+      name: '',
+      number: '',
+      checkpassword: '',
+    });
+    navigate('/login');
+    alert('입력하신 정보가 초기화되었습니다.');
+  };
+
+  // 폼 제출
+  const onSubmit = e => {
+    e.preventDefault();
+    // 로그인 API 호출
+    if (isLogin) {
+      fetch('http://10.58.52.82:3000/users/login', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+        },
+        body: JSON.stringify({
+          email: formData.id,
+          password: formData.password,
+        }),
+      })
+        .then(response => response.json())
+        .then(data => {
+          if (data.accessToken) {
+            localStorage.setItem('token', data.accessToken);
+            navigate('/');
+          } else {
+            alert('다시 로그인 해주세요.');
+            navigate('/login');
+          }
+        })
+        .catch(error => {
+          console.error(error);
+        });
+    } else {
+      // 회원가입 API 호출
+      if (formData.password !== formData.checkpassword) {
+        alert('비밀번호와 확인비밀번호가 일치하지 않습니다.');
+        return;
+      }
+      fetch('http://10.58.52.82:3000/users/signup', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json;charset=utf-8',
+        },
+        body: JSON.stringify({
+          email: formData.id,
+          name: formData.name,
+          password: formData.password,
+          phoneNumber: formData.number,
+        }),
+      })
+        .then(response => response.json())
+        .then(data => {
+          alert('회원가입이 성공적으로 완료되었습니다.');
+          navigate('/login');
+        })
+        .catch(error => {
+          console.error(error);
+        });
+    }
+  };
+
+  useEffect(() => {
+    setIsLogin(title === '로그인');
+  }, [title]);
+
+  return (
+    <S.LoginModal>
+      <GlobalStyle />
+      <S.LoginBox>
+        <S.LoginTitle>{title}</S.LoginTitle>
+        <S.Greet>휘게실에 오신 것을 환영합니다.</S.Greet>
+        <S.Form onSubmit={onSubmit}>
+          {(title === '로그인' ? LOGIN_INFO_DATA : SIGNUP_INFO_DATA).map(
+            info => {
+              const { id, name, type, placeholder } = info;
```

Comment:
```suggestion
            ({ id, name, type, placeholder } ) => {
```
여기도 이렇게 구조분해 할당을 해줄 수 있습니다!